packages/address/src/address.ts

```diff
@@ -0,0 +1,71 @@
+import { Logger } from '@ethersproject/logger';
+import { sha256 } from '@ethersproject/sha2';
+import { AbstractAddress } from '@fuel-ts/interfaces';
+import type { Bech32Address, B256Address } from '@fuel-ts/interfaces';
+
+import {
+  normalizeBech32,
+  isBech32,
+  toB256,
+  getBytesFromBech32,
+  toBech32,
+  getRandomB256,
+} from './utils';
+
+const logger = new Logger(process.env.BUILD_VERSION || '~');
+
+export default class Address extends AbstractAddress {
+  readonly bech32Address: Bech32Address;
+
+  constructor(address: string) {
+    super();
+    logger.checkNew(new.target, Address);
+    this.bech32Address = normalizeBech32(address);
+
+    if (!isBech32(this.bech32Address)) {
+      logger.throwArgumentError('Invalid Bech32 Address', 'address', address);
+    }
+  }
+
+  get address(): Bech32Address {
+    return this.bech32Address;
+  }
+
+  get b256Address(): B256Address {
```

Comment:
I think it would be good to export a `toB256 ` and `toBytes` instead of setting getting properties.